compiler/record-minmax/src/RecordMinMax.cpp

```diff
@@ -152,6 +161,50 @@ void RecordMinMax::initialize(const std::string &input_model_path)
   _interpreter->attachObserver(_observer.get());
 }
 
+// input_data_path is a text file which specifies the representative data
+// The text file should contain absolute file path per line.
+// The pointed file should be a binary file containing one representative input,
+// ready to be consumed by the input circle model without any modification
+void RecordMinMax::profileRawData(const std::string &mode, const std::string &input_data_path,
+                                  float min_percentile, float max_percentile)
+{
+  std::ifstream input_file(input_data_path);
+  if (input_file.fail())
+    throw std::runtime_error("Cannot open file \"" + input_data_path + "\".\n");
+
+  std::string record;
+  uint32_t num_records = 0;
+  const auto input_nodes = loco::input_nodes(_module->graph());
+  const auto num_inputs = input_nodes.size();
+
+  while (getline(input_file, record))
+  {
+    std::cout << "Recording " << num_records << "'th data" << std::endl;
+
+    uint32_t offset = 0;
+    for (auto input : input_nodes)
+    {
+      const auto *input_node = loco::must_cast<const luci::CircleInput *>(input);
+      std::vector<char> input_data(getTensorSize(input_node));
+      readDataFromFile(record, input_data.data() + offset, input_data.size());
+      _interpreter->writeInputTensor(input_node, input_data.data() + offset, input_data.size());
+
+      offset += input_data.size();
```

Comment:
This seems somewhat strange... offset accumulates and then read to `input_data.data() + offset`...
If there are 10 inputs, offset may go out of bounds.

Reply:
Ah, that was wrong implementation.. Thanks for pointing it out. I've updated the logic.